datafusion/src/datasource/object_store/mod.rs

```diff
@@ -141,7 +141,7 @@ pub trait ObjectStore: Sync + Send + Debug {
         let suffix = suffix.to_owned();
         Ok(Box::pin(file_stream.filter(move |fr| {
             let has_suffix = match fr {
-                Ok(f) => f.path().ends_with(&suffix),
+                Ok(f) => f.path().ends_with(&suffix) && !should_filter_out(f.path()),
```

Comment:
can you combine both logic into a more cohesive helper function

Reply:
@Jimexist Thanks, have merged into one func